test/common/CollectiveArgs.cpp

```diff
@@ -41,23 +41,33 @@ namespace RcclUnitTesting
     this->numInputElements  = numInputElements;
     this->numOutputElements = numOutputElements;
     this->scalarTransport   = scalarTransport;
-    this->scalarMode        = scalarMode;
+    this->scalarMode        = optionalColArgs.scalarMode;
 
-    if (scalarMode != -1)
+    if (optionalColArgs.scalarMode != -1)
     {
       size_t const numBytes = DataTypeToBytes(dataType);
-      if (scalarMode == ncclScalarDevice)
+      if (optionalColArgs.scalarMode == ncclScalarDevice)
       {
         CHECK_CALL(this->localScalar.AllocateGpuMem(numBytes));
         CHECK_HIP(hipMemcpy(this->localScalar.ptr, scalarTransport.ptr + (globalRank * numBytes),
                             numBytes, hipMemcpyHostToDevice));
       }
-      else if (scalarMode == ncclScalarHostImmediate)
+      else if (optionalColArgs.scalarMode == ncclScalarHostImmediate)
       {
         CHECK_HIP(hipHostMalloc(&this->localScalar.ptr, numBytes, 0));
         memcpy(this->localScalar.ptr, scalarTransport.ptr + (globalRank * numBytes), numBytes);
       }
     }
+    if (funcType == ncclCollAllToAllv)
+    {
+      for (int i = 0; i < this->totalRanks*this->totalRanks; ++i)
+      {
+        this->optionalArgs.sendcounts[i] = optionalColArgs.sendcounts[i];
+        this->optionalArgs.sdispls[i] = optionalColArgs.sdispls[i];
+        this->optionalArgs.recvcounts[i] = optionalColArgs.recvcounts[i];
+        this->optionalArgs.rdispls[i] = optionalColArgs.rdispls[i];
+      }
```

Comment:
this->options = options; 
We don't want to have to keep adding more and more code if we add/modify OptionalArgs.